### PR TITLE
Adapt namespaces

### PR DESCRIPTION
**Pull request by andresailer**
Adapted namespaces, mostly by `using dd4hep::Class` statements, sometimes replacing namespaces in the code. 
Changed `LCDD& lcdd` to `Detector& theDetector`
LCGEO compiles, and together with the outstanding PR in AIDASoft/DD4hep#180 the tests are passing

FIXME:
- [ ] I don't understand how Segmentation::decoder was working in the past, that is a`const` function but the returned pointer was still used with non-const function. Using `const_cast` now, but maybe I am just not getting the write Segmentation class ? Though I already I tried two of them `dd4hep::Segmentation` and `dd4hep::DDSegmentation::Segmentation`

BEGINRELEASENOTES
- Adapt to changes in namespaces in DD4hep
ENDRELEASENOTES

### CODE REVIEW

**Comment by gaede** [comment]
@andresailer in which class/file did you have an issue with the segmentation ?

File: detector/calorimeter/SHcalSc04_Barrel_v04.cpp [hunk 1]
```diff
@@ -144,36 +160,36 @@ static Ref_t create_detector(LCDD& lcdd, xml_h element, SensitiveDetector sens)
...
+  printout( dd4hep::DEBUG,  "SHcalSc04_Barrel_v04", "TPC_outer_radius : %e   - Ecal_outer_radius: %e ", TPC_outer_radius , Ecal_outer_radius) ;
 
   validateEnvelope(Hcal_inner_radius, Hcal_outer_radius, Hcal_radiator_thickness, Hcal_chamber_thickness, Hcal_nlayers);
 
   Readout readout = sens.readout();
-  Segmentation seg = readout.segmentation();
+  dd4hep::Segmentation seg = readout.segmentation();
   
-  BitField64& encoder = *seg.decoder() ;
+  dd4hep::DDSegmentation::BitField64& encoder = const_cast<dd4hep::DDSegmentation::BitField64&>(*seg.decoder());
```
Comment (andresailer) [inline detector/calorimeter/SHcalSc04_Barrel_v04.cpp]: Here ist `const_cast`

File: detector/calorimeter/SHcalSc04_Barrel_v04.cpp [hunk 2]
```diff
@@ -213,24 +229,24 @@ static Ref_t create_detector(LCDD& lcdd, xml_h element, SensitiveDetector sens)
...
     }
     
     // check if we have a TiledLayerGridXY segmentation :
-    const DD4hep::DDSegmentation::TiledLayerGridXY* ts0 =
-      dynamic_cast<const DD4hep::DDSegmentation::TiledLayerGridXY*>(  &multiSeg->subsegmentation( encoder.getValue() ) ) ;
+    const dd4hep::DDSegmentation::TiledLayerGridXY* ts0 =
+      dynamic_cast<const dd4hep::DDSegmentation::TiledLayerGridXY*>(  &multiSeg->subsegmentation( encoder.getValue() ) ) ;
     
-    tileSeg = const_cast<DD4hep::DDSegmentation::TiledLayerGridXY*>( ts0 ) ;
+    tileSeg = const_cast<dd4hep::DDSegmentation::TiledLayerGridXY*>( ts0 ) ;
```
Comment (andresailer) [inline detector/calorimeter/SHcalSc04_Barrel_v04.cpp]: Here

File: detector/calorimeter/SHcalSc04_Barrel_v04.cpp [hunk 3]
```diff
@@ -213,24 +229,24 @@ static Ref_t create_detector(LCDD& lcdd, xml_h element, SensitiveDetector sens)
...
       
       for( auto s : multiSeg->subSegmentations() ){
-	const DD4hep::DDSegmentation::TiledLayerGridXY* ts =
-	  dynamic_cast<const DD4hep::DDSegmentation::TiledLayerGridXY*>( s.segmentation ) ;
+	const dd4hep::DDSegmentation::TiledLayerGridXY* ts =
+	  dynamic_cast<const dd4hep::DDSegmentation::TiledLayerGridXY*>( s.segmentation ) ;
 	
 	if( ts ) {
-	  tileSeg = const_cast<DD4hep::DDSegmentation::TiledLayerGridXY*>( ts ) ;
+	  tileSeg = const_cast<dd4hep::DDSegmentation::TiledLayerGridXY*>( ts ) ;
```
Comment (andresailer) [inline detector/calorimeter/SHcalSc04_Barrel_v04.cpp]: Here

File: detector/calorimeter/SHcalSc04_EndcapRing_v01.cpp [hunk 4]
```diff
@@ -86,16 +108,16 @@ static Ref_t create_detector(LCDD& lcdd, xml_h element, SensitiveDetector sens)
   Readout readout = sens.readout();
   Segmentation seg = readout.segmentation();
   
-  BitField64& encoder = *seg.decoder() ;
+  BitField64& encoder = const_cast<BitField64&>(*seg.decoder());
```
Comment (andresailer) [inline detector/calorimeter/SHcalSc04_EndcapRing_v01.cpp]: Here

File: detector/calorimeter/SHcalSc04_Endcaps_v01.cpp [hunk 5]
```diff
@@ -59,43 +79,43 @@ static Ref_t create_detector(LCDD& lcdd, xml_h element, SensitiveDetector sens)
...
+  double      HcalEndcap_outer_radius          = theDetector.constant<double>("HcalEndcap_outer_radius");
+  double      HcalEndcap_min_z                 = theDetector.constant<double>("HcalEndcap_min_z");
+  double      HcalEndcap_max_z                 = theDetector.constant<double>("HcalEndcap_max_z");
   
   Readout readout = sens.readout();
   Segmentation seg = readout.segmentation();
   
 
-  BitField64& encoder = *seg.decoder() ;
+  BitField64& encoder = const_cast<BitField64&>(*seg.decoder());
```
Comment (andresailer) [inline detector/calorimeter/SHcalSc04_Endcaps_v01.cpp]: Here

**Comment by gaede** [comment]
Markus has changed the signature of `Segmentation::decoder()` from `BitField64* decoder() const;` to `const BitField64* decoder() const;` this is why this extra const_cast is needed now.
The other ones both where there before.
Eventually, we should review both - might be a problem in the API ...

**Pull request merged**